packages/reporter/test/sarif-formatter/add-artifact.test.ts

```diff
@@ -0,0 +1,35 @@
+import { Project } from 'fixturify-project';
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+import { afterEach, beforeEach, describe, expect, test } from 'vitest';
+
+import { sarifFormatter } from '../../src';
+
+describe('Test sarif-formatter add-artifact', () => {
+  let project: Project;
+
+  beforeEach(() => {
+    project = project = new Project('foo', '0.0.0');
+  });
+
+  afterEach(() => {
+    project.dispose();
+  });
+
+  test('should add an artifact only if it does not exist, should merge roles and properties correctly', async () => {
+    const data = JSON.parse(
+      readFileSync(resolve(__dirname, '../fixtures/sarif-formatter/add-artifact.json'), 'utf-8')
+    );
+    const sarif = sarifFormatter(data);
+    const artifacts = getArtifacts(sarif);
+
+    expect(artifacts).toMatchSnapshot();
+  });
+});
+
+function getArtifacts(sarif: string): string {
+  const parsedSarif = JSON.parse(sarif);
+  const parsedArtifacts = parsedSarif.runs[0].artifacts;
+  const artifacts = JSON.stringify(parsedArtifacts, null, 2);
+  return artifacts;
+}
```

Comment:
This can like likely be made into a function that takes the sarif and the key path you want to lookup and then shared by all the test.

Reply:
done